python/tskit/trees.py

```diff
@@ -7852,27 +7856,42 @@ def worker(sub_windows):
     # NOTE: see older definition of divmat here, which may be useful when documenting
     # this function. See https://github.com/tskit-dev/tskit/issues/2781
     def divergence_matrix(
-        self, *, windows=None, samples=None, num_threads=0, mode=None
+        self,
+        *,
+        windows=None,
+        samples=None,
+        num_threads=0,
+        mode=None,
+        span_normalise=True,
     ):
         windows_specified = windows is not None
-        windows = [0, self.sequence_length] if windows is None else windows
-
+        windows = self.parse_windows(windows)
         mode = "site" if mode is None else mode
 
-        # NOTE: maybe we want to use a different default for num_threads here, just
-        # following the approach in GNN
+        # FIXME this logic should be merged into __run_windowed_stat if
+        # we generalise the num_threads argument to all stats.
         if num_threads <= 0:
```

Comment:
not `<= 1`?

Reply:
Subtle difference - you might want to do the work in *one* background worker thread. Zero or less means that the work without threads, but default (which simplifies debugging a lot)